### PR TITLE
fix: Fix the SPI OAuth service deployment

### PR DESCRIPTION
**Pull request by metlos**
This adds two things:

* Run the OAuth service in debug mode for more detailed logging.
* Modify the deployment to authenticate with devsandbox proxy.

Hopefully, this modification will enable the service to work with Devsandbox proxy authorization.

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]

File: hack/development-mode.sh [hunk 1]
```diff
@@ -15,6 +15,10 @@ $ROOT/hack/util-update-app-of-apps.sh $MY_GIT_REPO_URL development $MY_GIT_BRANC
 # this needs to be pushed to your fork to be seen by argocd
 $ROOT/hack/util-set-development-repos.sh $MY_GIT_REPO_URL development $MY_GIT_BRANCH
 
+# set the API server which SPI uses to authenticate users to empty string (by default) so that multi-cluster
```
Comment (scoheb) [inline hack/development-mode.sh:18]: does this need to be added to `preview.sh`?

File: components/spi/oauth-service-deployment-patch.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[
+  {
+    "op": "add",
+    "path": "/spec/template/spec/containers/0/env",
+    "value": [
+      {
+        "name": "API_SERVER",
+        "value": "https://api-toolchain-host-operator.apps.appstudio-stage.x99m.p1.openshiftapps.com:443"
```
Comment (Michkov) [inline components/spi/oauth-service-deployment-patch.json:8]: what will happen when value will be empty? Will it use default kube API?

Comment (sparkoo) [reply components/spi/oauth-service-deployment-patch.json:8]: yes

**Review by Michkov — APPROVED** [review]

**Review by sparkoo — APPROVED** [review]
I don't like hardcoded apiserver value, but we need this PR to debug the issue on current deployment.

File: hack/development-mode.sh (hunk 1 above)
Comment (metlos) [reply hack/development-mode.sh:18]: Good point, it's been added.

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**